src/transformers/trainer.py

```diff
@@ -1552,6 +1548,10 @@ def _save_checkpoint(self, model, trial, metrics=None):
         else:
             torch.save(rng_states, os.path.join(output_dir, f"rng_state_{local_rank}.pth"))
 
+        # Maybe delete some older checkpoints.
+        if self.is_world_process_zero():
+            self._rotate_checkpoints(use_mtime=True, output_dir=run_dir)
+
```

Comment:
Moved this at the end otherwise we end up with folders with just the RNG states.